rotkehlchen/db/dbhandler.py

```diff
@@ -3282,6 +3285,46 @@ def delete_asset_identifier(self, asset_id: str) -> None:
         self.conn.commit()
         self.update_last_write()
 
+    def replace_asset_identifier(self, source_identifier: str, target_asset: Asset) -> None:
+        """Replaces a given source identifier either both in the global or the local
+        user DB with another given asset.
+
+        May raise:
+        - UnknownAsset if the source_identifier can be found nowhere
+        - InputError if it's not possible to perform the replacement for some reason
+        """
+        globaldb = GlobalDBHandler()
+        globaldb_data = globaldb.get_asset_data(identifier=source_identifier, form_with_incomplete_data=True)  # noqa: E501
+
+        cursor = self.conn.cursor()
+        userdb_query = cursor.execute(
+            'SELECT COUNT(*) FROM assets WHERE identifier=?;', (source_identifier,),
+        ).fetchone()[0]
+
+        if userdb_query == 0 and globaldb_data is None:
+            raise UnknownAsset(source_identifier)
+
+        if globaldb_data is not None:
+            globaldb.delete_asset_by_identifer(source_identifier, globaldb_data.asset_type)
+
+        if userdb_query != 0:
+            # the tricky part here is that we need to disable foreign keys for this
+            # approach and disabling foreign keys needs a commit. So rollback is impossible.
+            # But there is no way this can fail. (famous last words)
+            cursor.executescript('PRAGMA foreign_keys = OFF;')
+            cursor.execute(
+                'DELETE from assets WHERE identifier=?;',
+                (target_asset.identifier,),
+            )
+            cursor.executescript('PRAGMA foreign_keys = ON;')
+            cursor.execute(
+                'UPDATE assets SET identifier=? WHERE identifier=?;',
+                (target_asset.identifier, source_identifier),
+            )
+
+        self.conn.commit()
+        self.update_last_write()
```

Comment:
shouldn't this be inside the if?

Reply:
hmm yes actually it should. At first I thought no, but the previous if is only for global db.

Reply:
Will do in another PR.